lib/models/LifecycleConfiguration.js

```diff
@@ -539,39 +530,59 @@ class LifecycleConfiguration {
     _parseExpiration(rule) {
         const expObj = {};
         const subExp = rule.Expiration[0];
-        if (!subExp.Date && !subExp.Days && !subExp.ExpiredObjDeleteMarker) {
+        if (!subExp.Date && !subExp.Days && !subExp.ExpiredObjectDeleteMarker) {
             expObj.error = errors.MalformedXML.customizeDescription(
                 'Expiration action does not include an action time');
             return expObj;
         }
+        const eodm = 'ExpiredObjectDeleteMarker';
+        if (subExp.Date && (subExp.Days || subExp[eodm]) ||
+        (subExp.Days && subExp[eodm])) {
+            expObj.error = errors.MalformedXML.customizeDescription(
+                'Expiration action includes more than one time');
+            return expObj;
+        }
         if (subExp.Date) {
             const isoRegex = new RegExp('^(-?(?:[1-9][0-9]*)?[0-9]{4})-' +
-                '(1[0-2]|0[1-9])-(3[0-1]|0[1-9]|[1-2][0-9])T(2[0-3]|[0-1]' +
-                '[0-9]):([0-5][0-9]):([0-5][0-9])(\.[0-9]+)?(Z|[+-](?:2[0-3]' +
-                '|[0-1][0-9]):[0-5][0-9])?$');
+                '(1[0-2]|0[1-9])-(3[0-1]|0[1-9]|[1-2][0-9])T00:00:00' +
+                '(.[0-9]+)?(Z)?$');
             if (!isoRegex.test(subExp.Date[0])) {
-                expObj.error = errors.MalformedXML.customizeDescription(
-                    'Date is not in proper ISO 8601 format');
+                expObj.error = errors.InvalidArgument.customizeDescription(
+                    'Date must be at midnight GMT in ISO 8601 format');
```

Comment:
I think we don't have to comply exactly with AWS spec on this one, because we can provide more flexibility, and this sounds more like an AWS technical constraint than a real spec (and our system can handle it without more work), so it should be fine to allow any valid date string (can be constrained to only accept UTC / ISO 8601 though)

Reply:
sounds good 😸